notebooks/additional-methods.ipynb

```diff
@@ -0,0 +1,589 @@
+    "    time_horizon_timesteps = time_horizon + 1\n",
+    "    baseline = baseline[:time_horizon_timesteps]\n",
+    "    baseline_discounted = get_discounted_curve(discount_rate, baseline)\n",
+    "    baseline_atm_cost = np.trapz(baseline_discounted)\n",
+    "\n",
+    "    if method == \"mc\":\n",
+    "        # The Moura-Costa method calculates the ton-year benefit of a delayed emission\n",
+    "        # as the ton-years of carbon storage outside of the atmosphere over the period\n",
+    "        # 0<=t<=delay. Moura-Costa ignores the atmospheric impact of post-storage re-emission.\n",
+    "        delay_timesteps = delay + 1\n",
+    "        scenario = np.concatenate(\n",
+    "            (np.full(delay_timesteps, -1), np.zeros(len(baseline) - delay_timesteps))\n",
+    "        )\n",
+    "        scenario = get_discounted_curve(discount_rate, scenario)\n",
+    "        benefit = -np.trapz(scenario[:delay_timesteps])\n",
+    "\n",
+    "    elif method == \"lashof\":\n",
+    "        # The Lashof method calculates calculates the ton-year benefit of an emission at t=delay\n",
+    "        # as the atmospheric cost that no longer occurs within the time horizon. This can also\n",
+    "        # be understood as the difference between the baseline atmospheric cost and the scenario\n",
+    "        # atmospheric cost, calculated over the period delay<=t<=time_horizon.\n",
+    "        scenario = np.concatenate((np.zeros(delay), baseline))[:time_horizon_timesteps]\n",
+    "        scenario = get_discounted_curve(discount_rate, scenario)\n",
+    "        benefit = baseline_atm_cost - np.trapz(scenario[delay:])\n",
+    "\n",
+    "    elif method == \"car\":\n",
+    "        scenario = np.concatenate((np.zeros(delay), baseline))[:time_horizon_timesteps]\n",
+    "        scenario = get_discounted_curve(discount_rate, scenario)\n",
+    "        benefit = (1 / time_horizon) * baseline_atm_cost * delay\n",
+    "\n",
+    "    elif method == \"qc\":\n",
+    "        scenario = np.concatenate((np.zeros(delay), baseline))[:time_horizon_timesteps]\n",
+    "        scenario = get_discounted_curve(discount_rate, scenario)\n",
+    "        benefit = np.trapz(baseline[: delay + 1])\n",
+    "\n",
+    "    else:\n",
+    "        raise ValueError(f\"No ton-year accounting method called {method}\")\n",
+    "\n",
+    "    return {\n",
+    "        \"parameters\": {\n",
+    "            \"method\": method,\n",
+    "            \"time_horizon\": time_horizon,\n",
+    "            \"delay\": delay,\n",
+    "            \"discount_rate\": discount_rate,\n",
+    "        },\n",
+    "        \"baseline\": baseline_discounted,\n",
+    "        \"scenario\": scenario,\n",
+    "        \"baseline_atm_cost\": baseline_atm_cost,\n",
+    "        \"benefit\": benefit,\n",
+    "        \"num_for_equivalence\": baseline_atm_cost / benefit,\n",
+    "    }"
+   ]
+  },
```

Comment:
ideally i think we should have all the code in `tonyear` and not reproduced separately here. That just opens the possibility of the underlying `tonyear` implementation and the method in the notebook diverging. if you want to show bits of code, maybe a codeblock using triple backticks (\```)?

Reply:
Resolved with Joe's help!